File: src/scripts/model/buildings/zones/zoneWithJobs.js
import {DevelopmentState, Zone} from './zone.js';
import config from "../../../config.js";

export class ZoneWithJobs extends Zone {
  /**
   * @type {Citizen[]}
   */
  workers = [];
  maxWorkers = 0;

  get availableJobs() {
    return this.maxWorkers - this.workers.length
  }

  /**
   * Returns an HTML representation of this object
   * @returns {string}
   */
  toHTML() {
    let html = super.toHTML();
    html += `<div class="info-heading">Workers (${this.workers.length}/${this.maxWorkers})</div>`;

    html += '<ul class="info-citizen-list">';
    for (const worker of this.workers) {
      html += worker.toHTML();
    }
    html += '</ul>';

    return html;
  }
}

File: CitySim.Simulation/Model/Buildings/Zones/ZoneWithJobs.cs
namespace CitySim.Simulation.Model.Buildings.Zones;

public class ZoneWithJobs : Zone
{
    public List<Citizen> Workers    { get; set; } = [];
    public int           MaxWorkers { get; set; }

    public int AvailableJobs => MaxWorkers - Workers.Count;

    public ZoneWithJobs(Tile tile, string type) : base(tile, type) { }
}
